drivers/modem/hl7800.c

```diff
@@ -311,7 +311,15 @@ static const struct mdm_control_pinconfig pinconfig[] = {
 #define PROFILE_LINE_2                                                         \
 	"S00:255 S01:255 S03:255 S04:255 S05:255 S07:255 S08:255 S10:255\r\n"
 
-#define SETUP_GPRS_CONNECTION_CMD "AT+KCNXCFG=1,\"GPRS\",\"\",,,\"IPV4V6\""
+#define ADDRESS_FAMILY_IPV4 "IPV4"
+#if defined(CONFIG_MODEM_HL7800_ADDRESS_FAMILY_IPV4V6)
```

Comment:
Personally from my limited experience I'd prefer
```c
#if IS_ENABLED(CONFIG_*)
```

rather than `#ifdef *` or `#if defined(*)` because `IS_ENABLED` is only true when the CONFIG_* is defined to 1, and the code will compile as intended if the user unintentionally or unknowingly defined `CONFIG_*` = false somewhere and expect the features to be disabled.

Please correct me if I'm wrong

Reply:
Using `IS_ENABLED` is a good habit, but in practice there should be no big difference if one uses `defined`. I recall that there is a check that makes sure user is not able to define `CONFIG_` options in source files. Also one cannot define `CONFIG_*=false` in prj.conf file because Kconfig will make sure that the boolean value needs to be integer.

Reply:
@jukkar Wouldn't `CONFIG_SOMETHING=n` in `prj.conf` or `someboard_defconfig` get compiled into `autoconf.h` as
```c
#define CONFIG_SOMETHING 0
```
, and `#ifdef CONFIG_SOMETHING` will then becomes true?

Reply:
Dunno, but you can certainly try and see what happens. We do not have many invocations of `#if IS_ENABLED()` in the header files atm, so it does not seem be a big concern right now.
`find . -name '*.h' | xargs grep '#if IS_ENABLED'`

Reply:
I checked my app's `autoconf.h` things that are specifically "n" defined in my `prj.conf` aren't being compiled, which is great. I guess I'm just too paranoid.

Reply:
Thanks for the discussion. Going to leave this as is.